src/coreclr/jit/gentree.cpp

```diff
@@ -26036,6 +26036,76 @@ bool GenTreeHWIntrinsic::OperIsBitwiseHWIntrinsic() const
     return Oper == GT_AND || Oper == GT_OR || Oper == GT_XOR || Oper == GT_AND_NOT;
 }
 
+//------------------------------------------------------------------------
+// OperIsEmbRoundingEnabled: Is this HWIntrinsic a node with embedded rounding feature.
+//
+// Return Value:
+//    Whether "this" is a node with embedded rounding feature.
+//
+bool GenTreeHWIntrinsic::OperIsEmbRoundingEnabled() const
+{
+#if defined(TARGET_XARCH)
+    NamedIntrinsic intrinsicId = GetHWIntrinsicId();
+
+    if (!HWIntrinsicInfo::IsEmbRoundingCompatible(intrinsicId))
+    {
+        return false;
+    }
+
+    size_t numArgs = GetOperandCount();
+    switch (intrinsicId)
+    {
+        // these intrinsics only have the embedded rounding enabled implementation.
+        case NI_AVX512F_AddScalar:
+        case NI_AVX512F_DivideScalar:
+        case NI_AVX512F_MultiplyScalar:
+        case NI_AVX512F_SubtractScalar:
+        case NI_AVX512F_SqrtScalar:
+        {
+            return true;
+        }
+
+        case NI_AVX512F_Add:
+        case NI_AVX512F_Divide:
+        case NI_AVX512F_Multiply:
+        case NI_AVX512F_Subtract:
+
+        case NI_AVX512F_Scale:
+
+        case NI_AVX512F_ConvertScalarToVector128Double:
+        case NI_AVX512F_ConvertScalarToVector128Single:
+#if defined(TARGET_AMD64)
+        case NI_AVX512F_X64_ConvertScalarToVector128Double:
+        case NI_AVX512F_X64_ConvertScalarToVector128Single:
+#endif // TARGET_AMD64
+        {
+            return numArgs == 3;
+        }
+
+        case NI_AVX512F_Sqrt:
+        case NI_AVX512F_ConvertToInt32:
+        case NI_AVX512F_ConvertToUInt32:
+        case NI_AVX512F_ConvertToVector256Int32:
+        case NI_AVX512F_ConvertToVector256Single:
+        case NI_AVX512F_ConvertToVector256UInt32:
+        case NI_AVX512F_ConvertToVector512Single:
+        case NI_AVX512F_ConvertToVector512UInt32:
+        case NI_AVX512F_ConvertToVector512Int32:
```

Comment:
Looks like we're missing:
	* `Avx512F.FusedMultiplyAdd` (`VFMADD###PD`, `VFMADD###PS`)
	* `Avx512F.FusedMultiplyAddNegated` (`VFNMADD###PD`, `VFNMADD###PS`)
	* `Avx512F.FusedMultiplyAddNegatedScalar` (`VFNMADD###SD`, `VFNMADD###SS`)
	* `Avx512F.FusedMultiplyAddScalar` (`VFMADD###SD`, `VFMADD###SS`)
	* `Avx512F.FusedMultiplyAddSubtract` (`VFMADDSUB###PD`, `VFMADDSUB###PS`)
	* `Avx512F.FusedMultiplySubtract` (`VFMSUB###PD`, `VFMSUB###PS`)
	* `Avx512F.FusedMultiplySubtractAdd` (`VFMSUBADD###PD`, `VFMSUBADD###PS`)
	* `Avx512F.FusedMultiplySubtractNegated` (`VFNMSUB###PD`, `VFNMSUB###PS`)
	* `Avx512F.FusedMultiplySubtractNegatedScalar` (`VFNMSUB###SD`, `VFNMSUB###SS`)
	* `Avx512F.FusedMultiplySubtractScalar` (`VFMSUB###SD`, `VFMSUB###SS`)
	* `Avx512F.ScaleScalar` (`VSCALEFSD`, `VSCALEFSS`)
	* `Avx512DQ.ConvertToVector256Single` (`VCVTQQ2PS`, `VCVTUQQ2PS`)
	* `Avx512DQ.ConvertToVector512Double` (`VCVTQQ2PD`, `VCVTUQQ2PD`)
	* `Avx512DQ.ConvertToVector512Int64` (`VCVTPD2QQ`, `VCVTPS2QQ`)
	* `Avx512DQ.ConvertToVector512UInt64` (`VCVTPD2UQQ`, `VCVTPS2UQQ`)

Reply:
Thanks for pointing out, I was mostly referring to the APIs in #73604, might miss some in other subsets.

One more question is: those FMA instructions, do they have approved API overloads with embedded rounding, I might be missing it somewhere, and do we want to cover all the mentioned APIs above in this PR?

Reply:
> do they have approved API overloads with embedded rounding

They were approved as part of the general review, it's the same shape and just with the rounding mode as the last parameter. I'm not sure off the top of my head which issue they were exactly approved under, as we had several and things got split apart in multiple areas.

> do we want to cover all the mentioned APIs above in this PR?

It'd be nice, IMO, to just get these few missing ones. The rest of the PR looks generally good to me, so I think if we can get these ones up, then we'll be done with the entire `embedded rounding` feature here.